crates/criticalup-core/src/download_server_client.rs

```diff
@@ -93,32 +93,67 @@ impl DownloadServerClient {
         self.cache_dir.join("keys.json")
     }
 
-    fn product_release_manifest_cache_path(&self, product: &str, release: &str) -> PathBuf {
+    async fn product_release_manifest_cache_path(&self, product: &str, release: &str) -> PathBuf {
         self.product_release_cache_path(product, release)
+            .await
             .join("manifest.json")
     }
 
-    fn product_release_cache_path(&self, product: &str, release: &str) -> PathBuf {
-        self.cache_dir
+    async fn product_release_cache_path(&self, product: &str, release: &str) -> PathBuf {
+        let old_cache_dir = self.cache_dir.join("artifacts").join(product).join(release);
+
+        let new_cache_dir = self
+            .cache_dir
             .join("artifacts")
             .join("products")
             .join(product)
             .join("releases")
-            .join(release)
+            .join(release);
+        fs::create_dir_all(new_cache_dir.clone()).await.unwrap();
+
+        // If an old cache exist, we move its contents.
+        if old_cache_dir.exists() && !old_cache_dir.as_os_str().is_empty() {
+            // We want to fail silently.
+            if let Ok(mut paths) = fs::read_dir(&old_cache_dir).await {
+                if let Ok(Some(path)) = paths.next_entry().await {
+                    tracing::info!(
+                        "Tidying deprecated cache. New cache is located at {}",
+                        new_cache_dir.display()
+                    );
+                    let file_name = path.file_name();
+                    let old_file_name = old_cache_dir.join(&file_name);
+                    let new_file_name = new_cache_dir.join(&file_name);
+                    // If for some reason, the files are not copied,
+                    // we just want to return the new cache early and delete the old one.
+                    let _ = fs::copy(old_file_name, new_file_name).await.map(|_| async {
```

Comment:
Please handle the possible error instead of silently dropping it. You should not attempt to subvert error handling.